include/openthread/netdiag.h

```diff
@@ -345,6 +355,81 @@ otError otThreadSendDiagnosticReset(otInstance         *aInstance,
                                     const uint8_t       aTlvTypes[],
                                     uint8_t             aCount);
 
+/**
+ * Get the vendor name string.
+ *
+ * @param[in]  aInstance      A pointer to an OpenThread instance.
+ *
+ * @returns The vendor name string.
+ *
+ */
+const char *otThreadGetVendorName(otInstance *aInstance);
+
+/**
+ * Get the vendor model string.
+ *
+ * @param[in]  aInstance      A pointer to an OpenThread instance.
+ *
+ * @returns The vendor model string.
+ *
+ */
+const char *otThreadGetVendorModel(otInstance *aInstance);
+
+/**
+ * Get the vendor sw version string.
+ *
+ * @param[in]  aInstance      A pointer to an OpenThread instance.
+ *
+ * @returns The vendor sw version string.
+ *
+ */
+const char *otThreadGetVendorSwVersion(otInstance *aInstance);
+
+/**
+ * Set the vendor name string.
+ *
+ * Requires `OPENTHREAD_CONFIG_NET_DIAG_ADD_API_TO_CHANGE_VENDOR_NAME_MODEL_SW_VERSION`.
```

Comment:
nit:
```suggestion
 * Requires `OPENTHREAD_CONFIG_NET_DIAG_VENDOR_NAME_MODEL_SW_VERSION`.
```

Reply:
Folding this comment, I realized that there is a dedicated `OPENTHREAD_CONFIG_NET_DIAG_ADD_API_TO_CHANGE_VENDOR_NAME_MODEL_SW_VERSION` flag and it's not a typo.